purchases/src/defaults/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -839,6 +840,61 @@ class Purchases internal constructor(
             purchasesOrchestrator.allowSharingPlayStoreAccount = value
         }
 
+    /**
+     * The preferred UI locale override for RevenueCat UI components.
+     * This affects both API requests and UI rendering.
+     *
+     * @return The preferred UI locale override, or null if using system default
+     */
+    val preferredUILocaleOverride: String?
+        @Synchronized get() = purchasesOrchestrator.preferredUILocaleOverride
+
+    /**
+     * Override the preferred UI locale for RevenueCat UI components at runtime.
+     * This affects both API requests and UI rendering.
+     *
+     * If the locale changes, this will automatically clear the offerings cache and trigger
+     * a background refetch to get paywall templates with the correct localizations.
+     *
+     * @param localeString The locale string (e.g., "es-ES", "en-US") or null to use system default
+     * @return true if cache was cleared and refetch triggered, false if locale unchanged or cache clear rate limited
+     */
+    fun overridePreferredUILocale(localeString: String?): Boolean {
```

Comment:
Would it make sense to make this a "static" method instead of tying it to the Purchases instance? That might make it easier to access later on... I guess it wouldn't be cleared at any point, but normally that doesn't happen either...

Reply:
@tonidero The iOS PR has it on the instance so _probably_ best if we match consistency here? 🤷‍♂️  But also... I totally see why static would work here as well 😇